gnssanalysis/gn_download.py

```diff
@@ -854,17 +854,29 @@ def download_product_from_cddis(
     end_epoch: _datetime,
     file_ext: str,
     limit: int = None,
-    long_filename: bool = False,
+    long_filename: bool = None,
     analysis_center: str = "IGS",
     solution_type: str = "ULT",
     sampling_rate: str = "15M",
     project_type: str = "OPS",
     timespan: _datetime.timedelta = _datetime.timedelta(days=2),
     if_file_present: str = "prompt_user",
 ) -> None:
-    """
-    Download the file/s from CDDIS based on start and end epoch, to the
-    provided the download directory (download_dir)
+    """Download the file/s from CDDIS based on start and end epoch, to the download directory (download_dir)
+
+    :param _Path download_dir: Where to download files
+    :param _datetime start_epoch: Start date/time of files to find and download
+    :param _datetime end_epoch: End date/time of files to find and download
+    :param str file_ext: Extension of files to download (e.g. SP3, CLK, ERP, etc)
+    :param int limit: Variable to limit the number of files downloaded, defaults to None
+    :param bool long_filename: Search for IGS long filename, if None use start_epoch to determine, defaults to None
```

Comment:
This really should be `True` to enable, `False` to disable.
Unless a third value is needed to differentiate other behaviour, in which case to maintain valid typing, it would need wrapping in an `Optional`, i.e. `long_filename: Optional[bool] = None`

Reply:
Put in `Optional[bool]`